### PR TITLE
Cleanup guild chat system

### PR DESCRIPTION
**Pull request by P3ridot**

### CODE REVIEW

File: plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java [hunk 1]
```diff
@@ -61,127 +66,75 @@ public void onChat(AsyncPlayerChatEvent event) {
...
+        if (sendMessageToGuildAllies(player, guild, message)) {
             return true;
         }
 
-        return sendMessageToGuildMembers(event, message, player, guild);
+        return sendMessageToGuildMembers(player, guild, message);
     }
 
     private void spy(Player player, String message) {
         String spyMessage = ChatColor.GOLD + "[Spy] " + ChatColor.GRAY + player.getName() + ": " + ChatColor.WHITE + message;
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java:77]: Add option to configure format of this

File: plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java [hunk 2]
```diff
@@ -61,127 +66,75 @@ public void onChat(AsyncPlayerChatEvent event) {
...
-        }
-
+    private void sendMessageToGuild(Guild guild, String message) {
         PandaStream.of(guild.getMembers())
                 .filterNot(member -> member.getCache().isSpy())
                 .forEach(member -> member.sendMessage(message));
     }
 
-    private String getPositionString(Option<User> userOption) {
+    private String getUserPosition(Option<User> userOption) {
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java:137]: Move this to UserManager for easy access in other places

File: plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java [hunk 3]
```diff
@@ -61,127 +66,75 @@ public void onChat(AsyncPlayerChatEvent event) {
...
 
-        return false;
+    private boolean sendMessageToAllGuilds(Player player, Guild guild, String message) {
+        return this.sendMessageToGuilds(player, guild, config.chatGlobalDesign.getValue(), config.chatGlobal, message, this.guildManager.getGuilds());
     }
 
-    private boolean sendMessageToAllGuilds(AsyncPlayerChatEvent event, String message, Player player, Guild guild) {
-        String allGuildsPrefix = config.chatGlobal;
-        int prefixLength = allGuildsPrefix.length();
+    private boolean sendMessageToGuilds(Player player, Guild playerGuild, String chatDesign, String prefix, String message, Collection<Guild> receivers) {
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java:104]: Player -> User

File: plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java [hunk 4]
```diff
@@ -28,174 +32,113 @@ public void onChat(AsyncPlayerChatEvent event) {
...
-            format = StringUtils.replace(format, "{TAG}", StringUtils.replace(config.chatGuild.getValue(), "{TAG}", user.getGuild().get().getTag()));
-            format = StringUtils.replace(format, "{POS}", StringUtils.replace(config.chatPosition.getValue(), "{POS}", getPositionString(this.userManager.findByPlayer(player))));
-        }
-        else {
-            format = StringUtils.replace(format, "{TAG}", "");
-            format = StringUtils.replace(format, "{POS}", "");
-        }
+        user.getGuild()
+                .peek(guild -> {
+                    formatter.register("{TAG}", StringUtils.replace(config.chatGuild.getValue(), "{TAG}", user.getGuild().get().getTag()));
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java:59]: user.getGuild() here is cringe

Comment (Rollczi) [reply plugin/src/main/java/net/dzikoysk/funnyguilds/listener/PlayerChat.java:59]: `user.getGuild().get().getTag()` -> `guild.getTag()` ?

File: plugin/src/main/java/net/dzikoysk/funnyguilds/user/UserManager.java [hunk 5]
```diff
@@ -201,6 +205,27 @@ public boolean playedBefore(String nickname, boolean ignoreCase) {
...
+        return pluginConfiguration.chatPositionMember;
+    }
+
+    public String getUserPosition(Option<User> userOption) {
+        if (userOption.isEmpty()) {
+            return "";
+        }
+
+        return this.getUserPosition(userOption.get());
+    }
```
Comment (Rollczi) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/user/UserManager.java:227]: to raczej nie tutaj

**Review by Rollczi — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/net/dzikoysk/funnyguilds/user/UserManager.java (hunk 5 above)
Comment (PinkLolicorn) [reply plugin/src/main/java/net/dzikoysk/funnyguilds/user/UserManager.java:227]: image

**Review by dzikoysk — APPROVED** [review]

**Pull request merged**